tripper/utils.py

```diff
@@ -133,3 +140,41 @@ def parse_literal(literal: "Any") -> "Literal":
                 pass
 
     raise ValueError("cannot parse {literal=}")
+
+
+def parse_object(object: "Union[str, Literal]") -> "Union[str, Literal]":
+    """Applies heuristics to parse RDF object `object` to an IRI or literal.
+
+    The following heuristics is performed (in the given order):
+    - If `object` is a Literal, it is returned.
+    - If `object` is a string and
+      - starts with a scheme, it is asumed to be an IRI and returned.
+      - can be converted to a float, int or datetime, return it
+        converted to a literal of the corresponding type.
+      - it is a valid n3 representation, return it as the given type.
+      - otherwise, return it as a xsd:string literal.
+    - Otherwise, raise an ValueError.
+
+    Returns
+        A string if `object` is considered to be an IRI, otherwise a
+        literal.
+    """
+    if isinstance(object, Literal):
+        return object
+    if isinstance(object, str):
+        if re.match(r"^[a-z]+://", object):
+            return object
+        types = {
+            int: XSD.integer,
+            float: XSD.double,
+            datetime.datetime.fromisoformat: XSD.dateTime,
+        }
+        for pytype, datatype in types.items():
+            try:
+                pytype(object)  # type: ignore
```

Comment:
will never reach 'float'.

Reply:
It actually does. It is tested for in tests/test_units.py

Reply:
Fair enough, but then I would rather prefer some construct like this, to separate out the nasty `except: pass` in a more readable function and then explicitly test the object for attributes.

```
def isnumber(num):
    try:
        float(num)
        return True
    except ValueError:
        return False
    
...
    if isnumber(str):
        if str.isdigit():
            datatype=XSD.integer
        else:
            datatype = XSD.double
    else:
        ...
        datatype = XSD.datetime
       
```